DSESwiftTests/DSESwiftTests.swift

```diff
@@ -0,0 +1,412 @@
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+    
+    func testInsertDataPoint() {
+        let sensorOptions = SensorOptions(meta: "", uploadEnabled: true, downloadEnabled: true, persist: true)
+        do{
+            let sourceName = "testSource"
+            let sensor = Sensor(name: "sensor1", sensorOptions: sensorOptions, userId: KeychainWrapper.stringForKey(KEYCHAIN_USERID)!, source: "testSource", dataType: "JSON", synced: false)
+            try DatabaseHandler.insertSensor(sensor)
+            
+            var sensors = [Sensor]()
+            sensors = DatabaseHandler.getSensors(sourceName)
+            XCTAssertEqual(sensors.count, 1)
+            
+            var dataPoint = DataPoint(sensorId: sensor.id, value: "String value", date: NSDate(), synced: false)
+            try DatabaseHandler.insertOrUpdateDataPoint(dataPoint)
+            
+            dataPoint = DataPoint(sensorId: sensor.id, value: "String value", date: NSDate(), synced: false)
+            try DatabaseHandler.insertOrUpdateDataPoint(dataPoint)
+            
+            let dataPoints = try! DatabaseHandler.getDataPoints(sensorId: sensor.id, startDate: NSDate().dateByAddingTimeInterval( -7 * 24 * 60 * 60), endDate: NSDate().dateByAddingTimeInterval(100), limit: 100, sortOrder: SortOrder.Asc)
+            XCTAssertEqual(dataPoints.count, 2)
+        }catch{
+            XCTFail("Exception was captured. Abort the test.")
+        }
+    }
+    
+    func testGetDataPointWithValidLimit() {
+        let sensorOptions = SensorOptions(meta: "", uploadEnabled: true, downloadEnabled: true, persist: true)
+        do{
+            let sourceName = "testSource"
+            let sensor = Sensor(name: "sensor1", sensorOptions: sensorOptions, userId: KeychainWrapper.stringForKey(KEYCHAIN_USERID)!, source: "testSource", dataType: "JSON", synced: false)
+            try DatabaseHandler.insertSensor(sensor)
+            
+            var sensors = [Sensor]()
+            sensors = DatabaseHandler.getSensors(sourceName)
+            XCTAssertEqual(sensors.count, 1)
+            
+            var dataPoint: DataPoint!
+            for (var i=0; i < 101; i++){
+                dataPoint = DataPoint(sensorId: sensor.id, value: "String value", date: NSDate(), synced: false)
+                try DatabaseHandler.insertOrUpdateDataPoint(dataPoint)
+            }
+            
+            
+            let dataPoints = try! DatabaseHandler.getDataPoints(sensorId: sensor.id, startDate: NSDate().dateByAddingTimeInterval( -7 * 24 * 60 * 60), endDate: NSDate().dateByAddingTimeInterval(10), limit: 80, sortOrder: SortOrder.Asc)
+            XCTAssertEqual(dataPoints.count, 80)
+        }catch{
+            XCTFail("Exception was captured. Abort the test.")
+        }
+    }
+    
+    func testGetDataPointWithEmptyStartAndEnd() {
```

Comment:
You can use Long instead of long, then you can pass null. It is a required functionality to query data points with only startDate, with only endDate, or neither of them.